src/daemon/rpmostreed-utils.cxx

```diff
@@ -609,3 +610,52 @@ rpmostreed_parse_revision (const char  *revision,
 out:
   return ret;
 }
+
+/* Throw an error if there exists systemd inhibitor locks in `block` mode only.
+ * Note: systemd 248 provides a `--check-inhibitors` option, but it also checks
+ * for inhibitors in `delay` mode, which isn't what we want. */
+gboolean
+check_sd_inhibitor_locks (GCancellable    *cancellable,
+                          GError         **error)
+{
+  GDBusConnection *connection = rpmostreed_daemon_connection ();
+  g_autoptr(GVariant) inhibitors_array_tuple = 
+    g_dbus_connection_call_sync (connection, "org.freedesktop.login1", "/org/freedesktop/login1",
+                                 "org.freedesktop.login1.Manager", "ListInhibitors",
+                                 NULL, (const GVariantType*)"(a(ssssuu))",
+                                 G_DBUS_CALL_FLAGS_NONE, -1, cancellable, error);
+  if (!inhibitors_array_tuple)
+    return glnx_prefix_error (error, "Checking systemd inhibitor locks");
+  else if (g_variant_n_children (inhibitors_array_tuple) < 1)
+    return glnx_throw (error, "LoadInhibitors returned empty tuple");
+  g_autoptr(GVariant) inhibitors_array =
+    g_variant_get_child_value (inhibitors_array_tuple, 0);
+
+  char *what;
+  char *who;
+  char *why;
+  char *mode;
```

Comment:
Probably better to initialize all those to `NULL` in case it triggers static analyzers.